openni2_camera/src/openni2_frame_listener.cpp

```diff
@@ -70,38 +72,38 @@ void OpenNI2FrameListener::onNewFrame(openni::VideoStream& stream)
 
     rclcpp::Time ros_now = node_->now();
 
-    if (!user_device_timer_)
+    if (!use_device_timer_)
     {
       image->header.stamp = ros_now;
 
-      RCLCPP_DEBUG(rclcpp::get_logger("openni2"), "Time interval between frames: %.4f ms", (float)((ros_now - prev_time_stamp_).nanoseconds() / 1e6));
+      RCLCPP_DEBUG(rclcpp::get_logger("openni2"), "Time interval between frames: %.4f ms",
+                   (float)((ros_now.nanoseconds() - prev_time_stamp_.nanoseconds()) / 1e6));
```

Comment:
this needs to be X.nanoseconds() - Y.nanoseconds() because if you switch between use_device_time and not, you will get an std::runtime_error that the clock sources are different